### DISCUSSION TITLE
Como saber se você é bom suficiente?

### DISCUSSION THREAD

**Discussion by mathvbarone** [frontendbr/forum#770] [Discussão]
Fala galera, beleza?

Estou numa fase da carreira onde procuro evoluir como profissional ajudando outros profissionais.

Eu odeio quebrar a cabeça com problemas, e tenho o habito de, assim que resolvo algo, repassar esse conhecimento para que outros não sofram o que eu sofri.

Já trabalhei como professor e gosto muito de ensinar, inclusive, ao refazer meu site, fiz questão de fazer ele com um blog incluso:

http://matheusbarone.com/blog/

Sempre achei genial a ideia do Centro de Treinamento de criar um esquema de mentoria, e gostaria de ajudar como mentor. 

Acontece que na minha cabeça eu não estou pronto o suficiente para ensinar, se comparado ao nível dos outros caras da comunidade.

Eu tenho medo de me candidatar a mentor, e passar vergonha caso meu pupilo pergunte algo que eu mão conheça, pois, teoricamente eu só deveria me candidatar a professor após ter um certo nível de conhecimento.

Atualmente eu conheço as seguintes tecnologias:

- HTML
- CSS (SASS)
- CSS Frameworks (Bulma, Bootstrap)
- Javascript (Design Patterns)
- Jquery
- Git
- Jekyll
- Taskrunners (Gulp, Npm)
- Photoshop
- Illustrator
- Scrum
- UX/UI
- Inglês

Bom, a pergunta que fica é: 

Como saber que você é bom o suficiente para passar de aluno a professor?

Desde já agradeço!

**Comment by woliveiras** [comment]
Fala @mathvbarone!

Fico muito feliz por você sentir vontade de ajudar o Training Center, é sinal que nossos objetivos estão sendo cumpridos e estamos fazendo mais gente querer compartilhar o que sabe através de mentoria.

Seguindo pro seu problema... Algo que é perigoso quando nos auto-avaliamos é a subjetividade das coisas que pensamos. O que é **ser bom?**

Eu, William, sou bom em bagunçar no escritório, encontrar uns Gifs lokos e alegrar a galera. Isso é certeza.

Mas, será que sou bom tecnicamente?

O que é ser bom tecnicamente?

São perguntas bem difíceis de se responder.

O importante aqui é: você diz saber x coisas, dessas x coisas você saberia ensinar elas? - isso não só no mentoria, digo até no trabalho se entrar alguém que não conheça isso.

Eu sei que, tecnicamente, eu consigo desenvolver em JavaScript no Client Side buscando boas práticas, performance, dentre várias outras palavrinhas bonitas :joy:. Mas quem garante que, mesmo assim, nunca vai aparecer alguém com alguma dúvida que **eu nunca tive**? 

A maior característica de alguém que se coloca no papel de ajudar outras pessoas ensinando deveria ser a de admitir quando não sabe algo. Portanto, esse tipo de coisa:

> Eu tenho medo de me candidatar a mentor, e passar vergonha caso meu pupilo pergunte algo que eu não conheça, pois, teoricamente eu só deveria me candidatar a professor após ter um certo nível de conhecimento

**DEVE** acontecer.

**Professores não sabem de tudo.**

Na nossa área então, nem se fala. Não tem como saber de tudo. Algo sempre vai passar batido por você e alguém vai vir perguntar e você não vai saber responder. É aí que entra a humildade de dizer para a pessoa: olha, eu não sei o que é isso, nunca mexi com tal coisa.

O importante é não fazer como algumas pessoas que inventam alguma história pra fingir que sabe e não descer da postura de caixa da sabedoria e acabam deixando a pessoa que busca ajuda com um conteúdo errado na cabeça (vide um professor que disse em uma aula que DOM é o *domínio* no JavaScript).

## Primeiros passos no compartilhamento de conhecimento

Alguns primeiros passos que você pode fazer antes de mentorar: 

* responder em fóruns
* escrever artigos
* palestrar em um meetup

Todos esses pontos são coisas de feedback rápido. As pessoas vão te agradecer ou te corrigir bem rápido e nisso você já consegue ir se preparando para o que virá quando for ajudar alguém (seja no Mentoria ou não).

Em nenhum dos casos você deve se colocar a frente de ninguém como detentor da verdade absoluta ou de todo conhecimento do universo. Você aprende tanto quanto quem te escuta enquanto você ensina. Encare tudo isso como mais uma maneira de aprender, além de ajudar as pessoas.

Sempre que vou responder algo desse tipo tento não enviesar a pessoa a só contribuir no Training Center. O TC é uma de milhares de comunidades que precisam de ajuda. Porém, no seu caso, é interessante dar uma olhada neste artigo: [Dicas para novos(as) mentores(as)](https://medium.com/trainingcenter/dicas-para-novos-as-mentores-as-9e87fa30b2b4).

Além de mentorar, existem diversas maneiras de contribuir com o TC: ajudando as pessoas no Slack, [gravando vídeo tutoriais](https://medium.com/trainingcenter/o-training-center-busca-pessoas-para-cria%C3%A7%C3%A3o-de-tutoriais-bb6587018276), [escrevendo artigos](https://medium.com/trainingcenter/qualquer-pessoa-pode-escrever-pedir-ou-mandar-um-artigo-no-blog-do-training-center-fde2979e8067), [fazendo code-review](https://github.com/training-center/code-review/), ajudando a melhorar nossos processos... Vish! É muita maneira de ajudar. :joy:

**Comment by mathvbarone** [comment]
Fala @woliveiras!

Cara, te admiro muito por essa e tantas outros iniciativas em prol da comunidade.

Eu concordo 100% com você, o meu medo na verdade é passar uma imagem de "O cara não sabe nem fazer, e já quer ensinar", sabe?

Dessas tecnologias que eu listei, acredito que todas eu consiga ensinar, no minimo a usar da forma que eu uso (que não sei dizer se é a forma mais correta de todas).

Vou ler o artigo que você enviou, e seguir as outras dicas.

Muito obrigado pela ajuda!

**Comment by woliveiras** [comment]
Esqueci algumas dicas para sobreviver neste nosso mundo louco: escute o feedback das pessoas que estiverem absorvendo o que você ensina.

Através dos feedback's você saberá se está conseguindo transmitir o que realmente quer falar e vai receber ótimas dicas de como melhorar.

Existem muitas pessoas boas que vão te ajudar a fazer isso.

## Não alimente os Trolls

Algumas pessoas só vão querer te colocar pra baixo por algum motivo qualquer que faz com que elas se sintam superiores a você ou simplesmente por que elas não entendem o por que você faz algo. Não julgue-as.

Nós não sabemos o que leva essas pessoas a serem/agirem assim e por isso não devemos contra atacar. No máximo tentar extrair alguma opinião boa dessas pessoas, se ao menos elas tiverem algo pra dizer de verdade e não estiverem sabendo se expressar.

Por isso não se sinta pra baixo se aparecer alguém lhe dizendo algo como: "que conteúdo bosta, ein cara".

Garanto pra você que para cada comentário escroto irão aparecer pelo menos dois comentários bons.

**Comment by mathvbarone** [comment]
ahahahaahah 

>Por isso não se sinta pra baixo se aparecer alguém lhe dizendo algo como: "que conteúdo bosta, ein cara".

Para esse tipo de comentário eu só pergunto se o cara escreveu algum artigo que possa me ajudar em algo...

Na maioria das vezes as pessoas só sentam, julgam, e esperam o circo pegar fogo.

Eu não ligo não, acredito que o filtro das criticas construtivas vs criticas toxicas deve estar ligado desde sempre.

**Comment by woliveiras** [comment]
> "O cara não sabe nem fazer, e já quer ensinar"

Isso pode acontecer, viu mano.

As vezes pegamos uma tecnologia nova, estudamos por algumas horas e pensamos: "CARALEO, preciso compartilhar isso! Isso é muito bom!".

Você não domina aquilo 100%, mas quer muito compartilhar sobre. Onde quer que você esteja compartilhando você pode deixar explícito sua intenção: eu aprendi x e quero compartilhar isso.

Basta que não pareça em seu conteúdo que você domina aquilo 100%, você pode deixar isso claro para as pessoas na sua escrita ou fala. Não tem problema algum.

Muita gente faz palestras de assuntos que não domina 100% por que curtiu tanto que deseja compartilhar isso pro mundo.

No caso do mentoria, o buraco é bem mais embaixo, pois não é só saber coisa técnica e muitas vezes você e a pessoa que você estiver mentorando podem criar um grupo de estudos para aprenderem algo juntos. Não existe problema nisso. É só seguir o conselho de sempre dizer: olha, não sei sobre isso, nunca usei ou nunca tive a oportunidade de usar, mas posso tentar te ajudar a aprender enquanto eu também aprendo junto contigo.

É tudo muito colaborativo. :grin:

**Comment by willianjusten** [comment]
O @woliveiras já deu uma aula explicando praticamente tudo aqui. Só confirmando, ninguém sabe tudo no mundo, o importante é você estar disposto a aceitar críticas (tem algumas muito boas e que ajudam) e também aceitar que vão ter trolls (aqueles que vão falar mal, só com a intenção de te atingir).

Eu tinha muito desse medo, não escrevia, pois tinha medo de escrever merda ou escrever coisa que todo mundo já sabia. Acabei descobrindo que sempre tem espaço para escrever, mesmo que o assunto já seja passado, sempre tem gente disposta a aprender.

Eu tinha medo de palestrar, "já pensou, falar errado no frente de um monte de gente?" e descobri que isso é normal, o importante é sempre treinar e pedir ajuda a amigos antes, para fazer bonito depois. E bom, palestrei em tudo que é evento no Brasil.

Eu tinha medo de falar em Inglês com **qualquer** pessoa. Achava que meu inglês era macarrônico e feio. Até descobrir que o mais importante é ser entendido e bom, hoje trabalho numa empresa internacional num time com americano, russo, polonês, espanhol, dentre outras nacionalidades. E todos nós sempre nos ajudamos e o medo se foi.

Então é isso cara, não existe o "momento" em que você é bom e pode fazer as coisas. O momento é agora, comece e se jogue, aceite as boas críticas e evolua com isso =)

Muito desse medo é dado pela Síndrome do Impostor, que realmente é um impecilho na vida de quase todo dev. Eu escrevi um pouco sobre o que você perguntou/falou aqui, se quiser dar uma olhada, tem esse [link](https://willianjusten.com.br/sou-bom-o-suficiente/).

**Comment by woliveiras** [comment]
Esse post é realmente muito bom!!!

Aquele outro, do Will, "O que ninguém diz para os iniciantes" também é muito nessa pegada e vale a leitura.

**Comment by kazzkiq** [comment]
Uma vez eu li um relato no Quora que acho que pode ajudar (se lembrar o link posto aqui), era mais ou menos assim:

> Se você consegue tomar decisões técnicas e explicar porque elas foram as melhores escolhas, então você já é bom naquilo.

Acho que esse relato é um ótimo método para fazer autoavaliação. Pense em uma tecnologia que você quer ensinar, agora imagine que alguém te pergunta porque ela é a melhor solução para algo. Você consegue explicar, de forma técnica, porque ela é melhor? Qual a diferença dela para as outras? Consegue dar exemplos práticos de como ela traz vantagens? Se sim, você provavelmente já tem um conhecimento bacana nessa tecnologia.

**Comment by suissa** [comment]
Não sei se vcs tb passaram por isso mas quando eu percebi que estava razoavelmente bom foi quando eu comecei a codar algo completo e quando fui rodar ele executou sem erros. Pois antes rodaria já esperando algum erro para resolver, mas isso foi há muitoosss anos atrás. 

Hoje já vi que estou beeeem melhor pq consigo refatorar o código dos meus alunos sem nem rodar o código para testá-lo e minha solução não dá nenhum erro, fora obviamente a carga teórica que agreguei durante anos estudando, programando e dando aulas.

Mas eu acho que o principal é quando você aprender algo vc conseguir explicar para outra pessoa, pois só assim vc terá a garantia que aprendeu.

**Comment by bernardodiasc** [comment]
Por isso fico perturbando devs e managers pra documentar o projeto e código! hahaha

Mas é sério, independente do valor que a documentação carrega por si só, ao programar levando em conta essa documentação, algo colateral acontece que é um melhor entendimento do que se está fazendo resultando em uma arquitetura mais bem feita.

**Comment by mahenrique94** [comment]
Acho que ser **bom** irá envolver muitas coisas como:

- Saber as vantagens e desvantagens de uma linguagem(Saber porque devo utilizar uma linguagem para uma necessidade).
- Ja ter passado por vários problemas e ter conseguido resolve-los.
- Ter experiência(Um desenvolvedor de 10 anos ja passou e precisou desenvolver muito mais coisas do que um desenvolvedor de 2 anos)

Mas afinal, aplicar um teste poderia verificar o nível de conhecimento ? No meu ponto de vista não, porque muitas coisas acabamos esquecendo por falta de uso, mas sabemos que existe algo para resolver aquilo, basta verificar na doc como usa-la novamente.

Acredito também que aprendemos de acordo com a necessidade do projeto, se um projeto não precisar de upload, portanto, não iremos ter que aprender a fazer um upload(exemplo simples mais poderia ser qualquer outra necessidade).

Abraçoos

**Comment by diessica** [comment]
ensine algo quando você **quiser ensinar** – não quando você for sênior, não quando você souber as palavras exatas para explicar, não quando você terminar aquele livro, não quando você fizer faculdade.

você não precisa da afirmação de NINGUÉM, sério. don't overthink it
